consensus/istanbul/backend/engine_test.go

```diff
@@ -98,14 +101,16 @@ func newBlockChain(n int, items ...interface{}) (*blockchain.BlockChain, *backen
 			genesis.Config.Governance.Reward.ProposerUpdateInterval = uint64(v)
 		case governanceMode:
 			genesis.Config.Governance.GovernanceMode = string(v)
+		case *ecdsa.PrivateKey:
+			key = v
 		}
 	}
 	nodeKeys = make([]*ecdsa.PrivateKey, n)
 	addrs = make([]common.Address, n)
 
 	var b *backend
 	if len(items) != 0 {
-		b = newTestBackendWithConfig(genesis.Config)
+		b = newTestBackendWithConfig(genesis.Config, key)
```

Comment:
Can you use 48-52 for keys? Do you need to pass `nodekey` specifically?
```go
// These variables are the global variables of the test blockchain.
var (
	nodeKeys []*ecdsa.PrivateKey
	addrs    []common.Address
)
```

Reply:
I revised newBlockchain because the `key` below is not configurable(it is needed at `Test_GossipSubPeerTargets`). I couldn't think up another way... do you have any suggestion..?
```
func newTestBackendWithConfig(chainConfig *params.ChainConfig, key *ecdsa.PrivateKey) (b *backend) {
	dbm := database.NewDBManager(&database.DBConfig{DBType: database.MemoryDB})
	key, _ := crypto.GenerateKey()
	if chainConfig.Governance.GovernanceMode == "single" {
		// if governance mode is single, set the node key to the governing node.
		chainConfig.Governance.GoverningNode = crypto.PubkeyToAddress(key.PublicKey)
	}
}
```